src/main/java/com/exasol/versionnumberprovider/ExasolVersionNumberProvider.java

```diff
@@ -28,7 +28,8 @@ public class ExasolVersionNumberProvider {
     private static List<ExasolVersionNumber> extractVersions(final List<Tag> allTags) {
         return allTags.stream() //
                 .map(Tag::getName) //
-                .filter(tag -> !tag.startsWith("latest")) //
+                .filter(tag -> !(tag.startsWith("latest") || tag.contains("prerelease") || tag.contains("alpha")
```

Comment:
I am skeptic about whether _not latest_ should be ored with the other conditions.
Should it rather be `! latest && (...)` with `...` being the other conditions connected with `or` and in parenthesis?

Additionally (if that's not too much effort) I would enjoy having a dedicated private filter method with a nice name like `isAllowed()` or `isValidVersion()`. 

And maybe we could turn all the `tag.contains()` into a precompiled regular expression? 

And isn't looking for a substring `rc` or `alpha` a little bit risky? 
Should we add the dash `-` or use a more conservative approach with a more narrow regexp?

Reply:
# Extracted method
# Separated latest check from other checks
# Introduced pre-compiled pattern for all pre-releases

Since the logic can't handle much else than vanilla versions, filtering aggressively is actually safer.